test/test_dispatch.py

```diff
@@ -267,37 +279,87 @@ def test_impl_only(self):
             lambda m: m.impl_t_t("foo", "autograd"),
             # m.impl("foo", torch::kAutogradCPU, [](const Tensor& x) { return x })
             lambda m: m.impl_t_t("foo", "autogradcpu")
-        ])
-        self.assertExpectedInline(r, '''\
+        ]).state
+        self.assertExpectedInline(state, '''\
 name: test::foo
 schema: (none)
 CPU: impl_t_t :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
 AutogradCPU: impl_t_t :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
 Autograd[alias]: impl_t_t :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
 catchall: impl_t_t :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+''')
+
+    def test_computed_table(self):
+        result = self.commute("foo", [
+            # m.def("foo", [](const Tensor & x) { return x })
+            lambda m: m.def_name_t_t("foo"),
+            # m.impl("foo", torch::kCPU, [](const Tensor & x) { return x })
+            lambda m: m.impl_t_t("foo", "cpu"),
+            # m.impl("foo", torch::kCUDA, [](const Tensor & x) { return x })
+            lambda m: m.impl_t_t("foo", "xla"),
+            # m.impl("foo", torch::kAutograd, [](const Tensor & x) { return x })
+            lambda m: m.impl_t_t("foo", "autograd"),
+            # m.impl("foo", torch::kAutogradCPU, [](const Tensor & x) { return x })
+            lambda m: m.impl_t_t("foo", "autogradcpu")
+        ])
+        state, table = result.state, result.table
+        self.assertExpectedInline(state, '''\
+name: test::foo
+schema: test::foo(Tensor _0) -> (Tensor _0)
+debug: registered at /dev/null:0
+alias analysis kind: CONSERVATIVE
+CPU: impl_t_t :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+XLA: impl_t_t :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+AutogradCPU: impl_t_t :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+Autograd[alias]: impl_t_t :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+catchall: default_def_name_t_t :: (Tensor _0) -> (Tensor _0) [ boxed unboxed ]
+''')
+
+        def extract_entries(table, dispatch_keys):
+            extracted = ''
+            table_entries = table.split('\n')
+            for k in dispatch_keys:
+                for t in table_entries:
+                    if t.startswith(k):
+                        extracted += (t + '\n')
+            return extracted
```

Comment:
You're probably going to want to use this in other entries later.

Reply:
Yup! will move it out when it's used for the second time :D